app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :users, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'
+
+  after_create :update_posts_counter
+
```

Comment:
- [ ] [OPTIONAL] Hi, I noticed that you used `after_create `to call the `#update_post_counter` method in the Posts model. This means that the method will only be executed when a new post is created, but not when an existing post is updated. This can lead to inaccurate posts_count for the author if a post is edited or deleted later. A better option is to use `after_save`, which will run the method every time a post is saved to the database, whether it is created or updated. This way, the `posts_count` will always reflect the current number of posts for each author.
To use` after_save`, you just need to replace with `after_save` in the Posts model. For example, instead of writing this:

```rb
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  # ...
  after_create :update_posts_counter
  # ...
end
```
You can write this:

```rb
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  # ...
  after_save :update_posts_counter
  # ...
end
```

- [ ] [OPTIONAL] I also suggest that you make the update_post_counter method private, because it is not meant to be called from outside the model. Making it private will prevent accidental or malicious invocation of the method by other classes or objects. To make it private, you just need to add the keyword private before the method definition in the Posts model. For example, instead of writing this:

```rb

class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  # ...
  def update_posts_counter
    author.increment!(:posts_count)
  end
  # ...
end

```

```rb

class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  # ...
  private

  def update_posts_counter
    author.increment!(:posts_count)
  end
  # ...
end


```